src/snapshots/occlum/unionfs.rs

```diff
@@ -0,0 +1,167 @@
+    if !create_path.exists() {
+        fs::create_dir_all(create_path.as_path())?;
+    }
+
+    Ok(())
+}
+
+fn create_environment(mount_path: &Path) -> Result<()> {
+    let mut from_paths = Vec::new();
+    let mut copy_options = dir::CopyOptions::new();
+    copy_options.overwrite = true;
+
+    // copy the libs required by occlum to the mount path
+    from_paths.push(PathBuf::from("/lib64/ld-linux-x86-64.so.2"));
+    let mount_path_lib64 = mount_path;
+    let path_lib64 = mount_path_lib64.join("lib64");
+    create_dir(&path_lib64)?;
+
+    fs_extra::copy_items(&from_paths, &path_lib64, &copy_options)?;
+    from_paths.clear();
+
+    let mount_path_opt = mount_path;
+    let path_opt = mount_path_opt
+        .join("opt")
+        .join("occlum")
+        .join("glibc")
+        .join("lib");
+    create_dir(&path_opt)?;
+
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libc.so.6"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libdl.so.2"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libm.so.6"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libpthread.so.0"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/libresolv.so.2"));
+    from_paths.push(PathBuf::from("/opt/occlum/glibc/lib/librt.so.1"));
+    fs_extra::copy_items(&from_paths, &path_opt, &copy_options)?;
+    from_paths.clear();
+
+    create_dir(&mount_path.join("host"))?;
+    create_dir(&mount_path.join("proc"))?;
+    create_dir(&mount_path.join("dev"))?;
+    create_dir(&mount_path.join("etc"))?;
+    create_dir(&mount_path.join("lib"))?;
+    create_dir(&mount_path.join("root"))?;
+    create_dir(&mount_path.join("sys"))?;
+    create_dir(&mount_path.join("tmp"))?;
+
+    Ok(())
+}
+
+impl Snapshotter for Unionfs {
+    fn mount(&mut self, layer_path: &[&str], mount_path: &Path) -> Result<MountPoint> {
+        // From the description of https://github.com/occlum/occlum/blob/master/docs/runtime_mount.md#1-mount-trusted-unionfs-consisting-of-sefss ,
+        // the source type of runtime mount is "unionfs".
+        let fs_type = String::from("unionfs");
+        let source = Path::new(&fs_type);
+
+        if !mount_path.exists() {
+            fs::create_dir_all(mount_path)?;
+        }
+
+        // store the rootfs in different places according to the cid
+        let cid = mount_path
+            .parent()
+            .ok_or(anyhow!("parent do not exist"))?
+            .file_name()
+            .ok_or(anyhow!("Unknown error: file name pasre fail"))?;
+        let sefs_base = Path::new("/images").join(cid).join("sefs");
+        let unionfs_lowerdir = sefs_base.join("lower");
+        let unionfs_upperdir = sefs_base.join("upper");
+
+        let options = format!(
+            "lowerdir={},upperdir={},key={}",
+            unionfs_lowerdir.display(),
+            unionfs_upperdir.display(),
+            "c7-32-b3-ed-44-df-ec-7b-25-2d-9a-32-38-8d-58-61"
```

Comment:
Could you make this a global `const` with a comment explaining what this actually is (random uuid, or a magic/special value?)